src/System.Drawing.Common/src/System/Drawing/Icon.Unix.cs

```diff
@@ -322,9 +322,9 @@ public static Icon ExtractAssociatedIcon(string filePath)
             if (filePath == null)
                 throw new ArgumentNullException(nameof(filePath));
             if (string.IsNullOrEmpty(filePath))
-                throw new ArgumentException("Null or empty path.", "path");
+                throw new ArgumentException(SR.NullOrEmptyPath, "path");
```

Comment:
```suggestion
                throw new ArgumentException(SR.NullOrEmptyPath, nameof(filePath));
```

Reply:
what about backward compatibility? what if someone use
`catch (ArgumentException ex) where (ex.Message == "Null or empty path.")`

Reply:
You're correct in part, as apparently the Windows version would raise the ArgumentException against `path`, but only because it relies on `System.IO.Path.GetFullPath` for that behavior.

Though I'm not sure why changing it from `"path"` to `nameof(filePath)` breaks the scenario you are raising, because your scenario is already broken:
```
> new ArgumentException("Null or empty path.", "path").Message
"Null or empty path.\r\nParameter name: path"
> new ArgumentException("Null or empty path.", "filePath").Message
"Null or empty path.\r\nParameter name: filePath"
> 
```

Reply:
@watfordgnf, changing parameter name would be a breaking change, and it should be reviewed in some form (Certainly, it is out of scope of this PR; consider opening a new issue about ObjectDisposedExceptions and ParamNames). It would break the scenarios where exceptions are filtered by the parameter name, for example:
```cs
catch (ArgumentException ex) when (ex.ParamName == "path") { /* ... */}
```

Not sure how it became "Icon" / "path"  in the first place though. It doesn't make much sense... looking at the .NET Framework source, they do not seem to include ParamName at all, so it must be something from Mono.

Reply:
Do many people filter that way? We have not normally been concerned about changing the parameter name if it is wrong. At least, in.NET Core.

Reply:
I suppose it isn't very common.... but it certainly is a behavioural change. If it isn't a concern then I think it is right to use the correct paramname. 

Probably it isn't common to change `ParamName` in the first place, given that changing parameter name in itself breaks source compatibility (named parameter), and therefore you won't change it except cases like this where the `ParamName` was wrong, though.

Reply:
Yes we consider changing the actual parameter name to be breaking.